tests/lib/muinstaller/main.go

```diff
@@ -385,26 +385,43 @@ func createClassicRootfsIfNeeded(rootfsCreator string) error {
 	return nil
 }
 
-func createSeedOnTarget(bootDevice, seedLabel string) error {
-	// XXX: too naive?
-	dataMnt := runMntFor("ubuntu-data")
-	src := dirs.SnapSeedDir
-	dst := dirs.SnapSeedDirUnder(dataMnt)
-	// Remove any existing seed on the target fs and then put the
-	// selected seed in place on the target
-	if err := os.RemoveAll(dst); err != nil {
-		return err
-	}
+func createSeedOnTarget(bootDevice, src, dst string) error {
 	if err := os.MkdirAll(filepath.Dir(dst), 0755); err != nil {
 		return err
 	}
-	if output, err := exec.Command("cp", "-a", src, dst).CombinedOutput(); err != nil {
+	// Note that we do not use the -a option as cp returns an error if trying to
+	// preserve attributes in a fat filesystem. And this is fine for files from
+	// the seed, that do not need anything too special in that regard.
+	if output, err := exec.Command("cp", "-r", src, dst).CombinedOutput(); err != nil {
 		return osutil.OutputErr(output, err)
 	}
 
 	return nil
 }
 
+func createSeedOnSeedPartition(bootDevice string) error {
+	dst := runMntFor("ubuntu-seed")
+	for _, subDir := range []string{"snaps", "systems"} {
+		src := filepath.Join(dirs.SnapSeedDir, subDir)
+		if err := createSeedOnTarget(bootDevice, src, dst); err != nil {
```

Comment:
(nitpick) maybe just `copy()` or `copySeedDir()` ?  I stumbled a bit on the word "create" because it read a bit like there is some actual creation of a seed involved which it is not.

Reply:
Good point, changed now (also there was an unneeded argument that I didn't remove).